pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -194,9 +195,25 @@ def stop_at(obj):
                 # The Rot gate must be treated separately
                 if isinstance(current_gate, qml.Rot):
                     cumulative_angles = fuse_rot_angles(cumulative_angles, next_params)
+                    # For the Rot gate, the angles can cancel in a non-trivial way
+                    # e.g. Rot(φ,0,-φ) = RZ(φ) RY(0) RZ(-φ) = RZ(0) = I.
+                    angles_cancel = qml.math.allclose(
+                        qml.math.stack(
+                            [cumulative_angles[0] + cumulative_angles[2], cumulative_angles[1]]
+                        ),
+                        0.0,
+                        atol=atol,
+                        rtol=0,
+                    )
                 # Other, single-parameter rotation gates just have the angle summed
                 else:
                     cumulative_angles = cumulative_angles + next_params
+                    angles_cancel = qml.math.allclose(
+                        cumulative_angles,
+                        0.0,
+                        atol=atol,
+                        rtol=0,
+                    )
```

Comment:
Maybe just for readability, we could also do
```suggestion
                    # For the Rot gate, the angles can cancel in a non-trivial way
                    # e.g. Rot(φ,0,-φ) = RZ(φ) RY(0) RZ(-φ) = RZ(0) = I.
                    test_angles = qml.math.stack(
                        [cumulative_angles[0] + cumulative_angles[2], cumulative_angles[1]]
                    ),
                # Other, single-parameter rotation gates just have the angle summed
                else:
                    cumulative_angles = cumulative_angles + next_params
                    test_angles = cumulative_angles
                angles_cancel = qml.math.allclose(test_angles, 0.0, atol=atol, rtol=0)
```

Reply:
I like this 🙂 , thanks for the suggestion!